src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/prison/UserController.kt

```diff
@@ -137,6 +137,53 @@ class UserSearchController(
     ),
   )
 
+  @PreAuthorize("hasRole('USERS__PRISON_USERS__FIND_BY_CASELOAD_AND_ROLE__RO')")
+  @GetMapping("/prisonusers/find-by-caseload-and-role")
+  @Operation(
+    summary = "Get all users filtered by active caseload and role",
+    description = "Requires role USERS__PRISON_USERS__FIND_BY_CASELOAD_AND_ROLE__RO<br/>Get all users with active caseload and nomis role.<br/> This search does not limit by user token.",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Pageable list of user summaries",
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Incorrect filter supplied",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  fun getUsersByCaseloadAndRole(
+    @PageableDefault(sort = ["lastName", "firstName"], direction = Sort.Direction.ASC)
+    pageRequest: Pageable,
+    @Parameter(
+      description = "Filter results by user's currently active caseload i.e. the one they have currently selected",
+      example = "MDI",
+    )
+    @RequestParam(value = "activeCaseload", required = true)
+    activeCaseload: String,
+    @RequestParam(value = "nomisRole", required = true)
+    @Parameter(
+      description = "Filter will match users that have the NOMIS role specified",
+      example = "201",
```

Comment:
this should be a role that is used in DPS not a numeric role code thats only used in NOMIS - EG KEYWORKER_MONITOR woudl be a better example